### PR TITLE
Testing Quality of Life Improvements

### PR DESCRIPTION
**Pull request by greggroth**
As someone new to typescript and not very familiar with the tooling and best practices, I struggled to get up-and-running with the project 😅 . I've included configuration for the Mocha Sidebar VS Code extension that is [recommended by the Mocha project](https://mochajs.org/#mocha-sidebar-vs-code) and simple instructions in the README for getting a green test suite after cloning or opening the project in a Codespace.

### CODE REVIEW

**Review by joshspicer — APPROVED** [review]
Very helpful, thanks!

**Review by chrmarti — CHANGES_REQUESTED** [review]
Thanks! Left a comment regarding prettier.

File: package.json [hunk 1]
```diff
@@ -81,7 +83,9 @@
 		"gulp-filter": "^7.0.0",
 		"mocha": "^9.2.1",
 		"npm-run-all": "^4.1.5",
+		"nyc": "^15.1.0",
 		"p-all": "^4.0.0",
+		"prettier": "^2.8.0",
```
[the remark this answers is not in the document]
Comment (greggroth) [reply package.json:88]: It's opinionated, but configurable https://github.com/devcontainers/cli/pull/312/files#diff-663ade211b3a1552162de21c4031fcd16be99407aae5ceecbb491a2efc43d5d2

I don't have a strong feeling about which formatter is used if you prefer another, but I think having one would be beneficial. For example, as I was working in the codebase I found I was fighting with VS Code to match the existing code format. Use Prettier means contributors don't have to think about that.

[the remark this answers is not in the document]
Comment (greggroth) [reply package.json:88]: Here's an example of two json files with completely different indentations (tabs v spaces and size):

https://github.com/devcontainers/cli/blob/main/.vscode/settings.json
https://github.com/devcontainers/cli/blob/main/src/test/configs/image-metadata/.devcontainer/devcontainer.json

**Comment by greggroth** [comment]
Closing since it's stale. We can grab all or parts of this if there's still interest in the future.

**Pull request closed without merging**